raiden/network/proxies/user_deposit.py

```diff
@@ -101,19 +100,70 @@ def deposit(
         error_prefix = "Call to deposit will fail"
 
         with self.deposit_lock:
-            amount_to_deposit, log_details = self._deposit_preconditions(
-                total_deposit=total_deposit,
-                beneficiary=beneficiary,
-                token=token,
-                block_identifier=block_identifier,
-            )
+            # check preconditions
+            try:
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=given_block_identifier
+                )
+                current_balance = token.balance_of(
+                    address=self.node_address, block_identifier=given_block_identifier
+                )
+            except (BadFunctionCallOutput, ValueError):
+                # If 'given_block_identifier' has been pruned, we cannot perform the
+                # precondition checks but must still set the amount_to_deposit to a
+                # reasonable value.
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=checking_block
+                )
+                amount_to_deposit = TokenAmount(total_deposit - previous_total_deposit)
+            else:
+                log_details["previous_total_deposit"] = previous_total_deposit
+                amount_to_deposit = TokenAmount(total_deposit - previous_total_deposit)
+
```

Comment:
This is missing:
1. [Check for overflows](https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/data/source/services/UserDeposit.sol#L111)
2. [Check for deposit limit](https://github.com/raiden-network/raiden-contracts/blob/master/raiden_contracts/data/source/services/UserDeposit.sol#L115)

Reply:
Yes, I'll add them.